### DISCUSSION TITLE
Support for multiple managed assemblies that refrence `UnrealEngine.Framework`

### DISCUSSION THREAD

**Discussion by riddlemd** [nxrighthere/UnrealCLR#78] [General]
I've got some features that I'm building out in a separate projects that are used in multiple game prototypes. They all reference the UnrealEngine.Framework, but don't have a main entry point (They're stuff like inventory systems, etc.) this causes the actual game project's Main entry point to be ignored (due to, best I can guess, the supporting projects dll being alphabetically first.)

Any way, or plan to allow us to specify which dll to use as the game's primary dll?

**Comment by nxrighthere** [comment]
Please, see [this](https://github.com/nxrighthere/UnrealCLR/discussions/65) discussion, feel free to participate.

**Reply by riddlemd** [reply]
I'm not trying to load dll dynamically -- I simply want to have more than 1 dll with reference to UnrealEngine.Framework so I can split my codebase into multiple projects. Loading stuff dynamically is outside the scope of what I'm trying to do.

My issue seems to stim from the plugin only checking 1 of the DLL for a Main class and ignoring the others vs the discussion which seems to deal more with reflection and dynamic loading.

**Reply by nxrighthere** [reply]
> I'm not trying to load dll dynamically -- I simply want to have more than 1 dll with reference to UnrealEngine.Framework so I can split my codebase into multiple projects. Loading stuff dynamically is outside the scope of what I'm trying to do.

This is not how the runtime works. You can reference framework assembly only once. Anything else requires dynamic loading into the same assembly load context.
